src/graph/validator/GoValidator.cpp

```diff
@@ -183,6 +185,24 @@ void GoValidator::extractPropExprs(const Expression* expr,
   const_cast<Expression*>(expr)->accept(&visitor);
 }
 
+Expression* GoValidator::rewriteVertexEdge2EdgeProp(const Expression* expr) {
+  auto pool = qctx_->objPool();
+  const auto& name = expr->toString();
+  if (name == "id($^)" || name == "src(edge)") {
```

Comment:
Compare expression structure instead of string.

Reply:
why